r5dev/src/hooks/cnetmessage.cpp

```diff
@@ -0,0 +1,51 @@
+#include "pch.h"
+#include "hooks.h"
+
+namespace Hooks
+{
+	SVC_Print_ProcessFn originalSVC_Print_Process = nullptr;
+}
+
+static std::ostringstream oss_print;
+static auto ostream_sink_print = std::make_shared<spdlog::sinks::ostream_sink_st>(oss_print);
+
+//-----------------------------------------------------------------------------
+// Purpose: log the console messages sent by server
+//-----------------------------------------------------------------------------
+bool Hooks::SVC_Print_Process(__int64 thisptr)
+{
+	// TODO: replace with VT hook
```

Comment:
Eh, we won't use Virtual Table hooks.
We would need to use Virtual Method hooks because we hook into the game before it even initialized all class instances.

Reply:
Virtual Method hooking isn't really needed at the moment since we have JMP hooks.
If I deem it neccessary I will implement it at a later date.

Reply:
VTable hooks would work for some class instances that already have the virtual table assigned but some don't have it and get it via initializing the game.